### ISSUE TITLE
Allow appointments to be scheduled without confirmation

### ISSUE THREAD

**Issue by arcstur** [nextcloud/calendar#5325]
### Is your feature request related to a problem? Please describe.

For some specific use cases I don't mind allowing clients to schedule an appointment without them confirming in the email. I know this is important for them to receive the correct location and etc, but sometimes that's not needed.

### Describe the solution you'd like

When creating an appointment, the user can uncheck a box "Ask for email confirmation before scheduling", which is checked by default.

**Comment by wiredone** [comment]
I have updated the text in my page to this,
'Important:  Your booking will not be made until you confirm via the email that you will be sent.  In different words: your appointment will not be booked and it will not appear in my calendar and I will have no awareness of it until you click on the *confirm* link in the email.  Depending on your email service, it might take a few minutes to arrive.'

noting: https://github.com/nextcloud/calendar/issues/5189

**Comment by philiprenich** [comment]
Just ran in to this issue and would love this enhancement!

The person even emailed me to say "Thanks! I selected 9:30, talk then!" but they didn't actually click confirm in the email. I don't even know if they saw the confirmation email (could have gone to spam? just didn't check yet?). The act of scheduling the appointment is pretty explicit already so there's a strong use case to consider that workflow confirmation enough. 

All that's needed after they complete the schedule form is an email with the .ics invite and location link. Those could both be done in-browser as well!

**Comment by PhiCecHey** [comment]
I'd love this enhancement as well. Unfortunately, because of the issues described above, I can't use the Appointments feature until there's a solution for this. The people I'd share the link with are family, so they wouldn't spam my calendar and I don't mind deleting the occasional event by hand if it comes to that. Maybe the enhancement could also notify the calendar owner whenever a new appointment is scheduled.

**Comment by AlgopiG** [comment]
I agree fully that this feature is necessary. As i would want to link the appointment directly to a payment with Stripe , as long as there is a double confirmation by mail, no chance that a customer plays the entire sequence

**Comment by sunbeam60** [comment]
Completely agree this is a necessary option. I only share the secret invite links with people I trust, as a way to easy friction for an initial meeting. Instead this adds friction.

**Comment by tamarhayrikyan-tech** [comment]
Same issue. Started using NextCloud instead of Google for this, but on the first try with a third party it didn't work because they don't expect that extra step of opening the confirmation email and clicking the confirmation link.

**Comment by MrRinkana** [comment]
Another thing is that the appointment technically is to also be confirmed by the organiser, and any such confirmation (or declination) might be confusing after the first confirmation e-mail

**Comment by hubfc** [comment]
I'd like to suggest a middle ground between "email confirmation link" and "no confirmation at all". It fixes the friction described here and keeps spam protection.

**Use a regular calendar invitation as the confirmation step:**

1. The booker submits the form.
2. Nextcloud sends a normal iMIP invitation (`.ics`, `METHOD:REQUEST`) instead of the mail with the confirmation link.
3. The booker clicks "Accept" in whatever calendar they use (Outlook, Gmail, Apple Calendar, Thunderbird …). The appointment is in their calendar right away.
4. The accept reply confirms the booking and the event is added to the organizer's calendar. Until then the slot could be held as tentative.

**Why this helps:**

- Bookers already know this flow from other booking tools. The problems mentioned above (link not noticed, people thinking they have already booked) mostly come from the unfamiliar confirmation link, not from having a confirmation step at all.
- It's still spam-safe. Entering someone else's address doesn't fill the organizer's calendar, because the owner of that address has to accept.
- It goes in the direction @philiprenich described (just send the `.ics`), without dropping the protection completely.
- It could also help with @MrRinkana's point: a later organizer confirmation or decline could simply be sent as an update to the same invitation instead of another separate mail.

This could be offered per appointment type next to the "no confirmation" option, e.g. **Confirmation: email link / calendar invitation / none**. That would also cover the "two URLs with and without spam protection" idea from @wiredone.

One open question is how reliably Nextcloud processes replies from external attendees today. The flow depends on that working.

**Comment by samvallad33** [comment]
Allowing appointments to be scheduled without confirmation is a destructive auto-approve on user calendars. Scheduling mutations need an explicit human gate. Are you requiring confirmation before any appointment write is committed?

**Comment by SebastianKrupinski** [comment]
Hi @hubfc 

That is a great compromise! I will but this idea up to design to see what they think. My only worry is this changes the current flow, so users that are already used to it, might be confused

@nextcloud/designers @kra-mo

**Comment by philiprenich** [comment]
@hubfc @SebastianKrupinski If holding the slot tentatively is something that can actually be displayed in the UI, that would be good. That way, if the booker doesn't get all the way through their flow, the organiser could at least see an attempt was made and reach out to the person to confirm (useful for when you know the booker, maybe not for a hairdresser making many appointments with strangers)

**Comment by hubfc** [comment]
Thanks @SebastianKrupinski! glad you like it.

About the flow change: most people who book an appointment have nothing to do with Nextcloud. They've used Calendly or similar tools, where you just get a calendar invite as I roughly described. The confirmation link is what confuses them, and that's what most complaints here are about when I use it with others. Organizers who like the current flow could keep it as the default if you want to keep the current flow (but I know: probably more work and more code to maintain)

@philiprenich I'd drop the tentative part of my suggestion. If unconfirmed bookings block slots, anyone can fill a calendar with fake addresses. Showing pending bookings somewhere (a list or a notification) would give the organizer the same information without blocking anything. But this is nice to have, I guess.
